exchange/utils_test.go

```diff
@@ -1408,3 +1408,254 @@ func TestBidderToPrebidChainsZeroLengthSChains(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, len(output), 0)
 }
+
+func TestRemoveUnpermissionedEids(t *testing.T) {
+	bidder := "bidderA"
+
+	testCases := []struct {
+		description     string
+		userExt         json.RawMessage
+		eidPermissions  []openrtb_ext.ExtRequestPrebidDataEidPermission
+		expectedUserExt json.RawMessage
+	}{
+		{
+			description: "Extension Nil",
+			userExt:     nil,
+			eidPermissions: []openrtb_ext.ExtRequestPrebidDataEidPermission{
+				{Source: "source1", Bidders: []string{"bidderA"}},
+			},
+			expectedUserExt: nil,
+		},
+		{
+			description: "Extension Empty",
+			userExt:     json.RawMessage(`{}`),
+			eidPermissions: []openrtb_ext.ExtRequestPrebidDataEidPermission{
+				{Source: "source1", Bidders: []string{"bidderA"}},
+			},
+			expectedUserExt: json.RawMessage(`{}`),
+		},
+		{
+			description: "Extension Empty - Keep Other Data",
+			userExt:     json.RawMessage(`{"other":42}`),
+			eidPermissions: []openrtb_ext.ExtRequestPrebidDataEidPermission{
+				{Source: "source1", Bidders: []string{"bidderA"}},
+			},
+			expectedUserExt: json.RawMessage(`{"other":42}`),
+		},
+		{
+			description: "Eids Empty",
+			userExt:     json.RawMessage(`{"eids":[]}`),
+			eidPermissions: []openrtb_ext.ExtRequestPrebidDataEidPermission{
+				{Source: "source1", Bidders: []string{"bidderA"}},
+			},
+			expectedUserExt: json.RawMessage(`{"eids":[]}`),
+		},
+		{
+			description: "Eids Empty - Keep Other Data",
+			userExt:     json.RawMessage(`{"eids":[],"other":42}`),
+			eidPermissions: []openrtb_ext.ExtRequestPrebidDataEidPermission{
+				{Source: "source1", Bidders: []string{"bidderA"}},
+			},
+			expectedUserExt: json.RawMessage(`{"eids":[],"other":42}`),
+		},
+		{
+			description:     "Allowed By Nil Permissions",
+			userExt:         json.RawMessage(`{"eids":[{"source":"source1","id":"anyID"}]}`),
+			eidPermissions:  []openrtb_ext.ExtRequestPrebidDataEidPermission{},
+			expectedUserExt: json.RawMessage(`{"eids":[{"source":"source1","id":"anyID"}]}`),
+		},
+		{
+			description:     "Allowed By Empty Permissions",
```

Comment:
What is the difference between nil permissions and empty permissions?

Reply:
As of now, nothing. Good catch. This was a copy/paste error. Fixed.